rule/rule.go

```diff
@@ -50,6 +51,14 @@ type Match struct {
 	// The following regexp example matches all paths of the domain `mydomain.com`: `https://mydomain.com/<.*>`.
 	// The glob equivalent of the above regexp example is `https://mydomain.com/<*>`.
 	URL string `json:"url"`
+
+	// A map of HTTP headers. When ORY Oathkeeper searches for rules
+	// to decide what to do with an incoming request to the proxy server, it compares the HTTP headers of the incoming
+	// request with the HTTP headers of each rules. If a match is found, the rule is considered a partial match.
+	// For headers with values in array format (e.g. User-Agent headers), the rule header value must match at least one
+	// of the request header values.
+	// If the matchesUrl and matchesMethods fields are satisfied as well, the rule is considered a full match.
+	Headers map[string]string `json:"headers"`
```

Comment:
As far as I understand, this is an `AND` match. So if all requirements are met from the rule, it will apply. Is that desired? Are there use cases for `OR`? And if so, how do we handle those? We can also only support `AND` for now but will nede to support `OR` most likely at some point, without breaking the config.

Reply:
At this point, the implementation only supports `AND`.
You have to match all headers specified in the rule. IMHO, an `OR` implementation would be overcomplicating things. It should just be a different rule. Headers are seen just like URLs are (although, it is true that URLs support wildcards).

Reply:
I think that using using `OR` is definitely a breaking change as it would potentially override normal oathkeeper rule matching and documentation with an optional routing header. Therefore i would recommend supporting only `AND` for now as an advanced configuration parameter.

Reply:
Ok, that makes sense!